### PR TITLE
Move icu_plurals and icu_decimal to no_std

### PR DESCRIPTION
**Pull request by Manishearth**
Progress towards #812

### CODE REVIEW

File: components/plurals/src/operands.rs [hunk 1]
```diff
@@ -80,6 +79,9 @@ impl PluralOperands {
     /// Returns the number represented by this [`PluralOperands`] as floating point.
     /// The precision of the number returned is up to the representation accuracy
     /// of a double.
+    ///
+    /// This method requires the `"std"` feature be enabled
+    #[cfg(feature = "std")]
```
Comment (Manishearth) [inline components/plurals/src/operands.rs:84]: Worth highlighting: some of the math functions aren't in std

File: ffi/ecma402/Cargo.toml [hunk 2]
```diff
@@ -29,6 +29,7 @@ all-features = true
 ecma402_traits = { version = "0.2.0" }
 icu = { path = "../../components/icu", default-features = false }
 icu_provider = { version = "0.2", path = "../../provider/core" }
+icu_plurals = { version = "0.2", path = "../../components/plurals", features = ["std"] }
```
Comment (Manishearth) [inline ffi/ecma402/Cargo.toml:32]: I had to add this because ecma402_traits requires error types to implement `Error`

Comment (Manishearth) [reply ffi/ecma402/Cargo.toml:32]: Filed https://github.com/google/rust_icu/issues/207

**Review by sffc — APPROVED** [review]
The "n" function being behind std seems odd, but I think it's mostly not used

**Comment by Manishearth** [comment]
> The "n" function being behind std seems odd, but I think it's mostly not used

There's been discussion about the math functions being a part of std in the past; the contention is that baremetal targets may wish to plug in their own. We could include a no_std manual `powi()` function that's slower but workable if we really want, but that function isn't used much so I didn't bother.

**Review by sffc — APPROVED** [review]

**Pull request merged**